lib/galaxy/jobs/runners/slurm.py

```diff
@@ -36,37 +50,43 @@ def __get_jobinfo():
             if p.returncode != 0:
                 # Will need to be more clever here if this message is not consistent
                 if stderr == 'slurm_load_jobs error: Invalid job id specified\n':
-                    return dict( JobState='NOT_FOUND' )
+                    # The job may be old, try to get its state with sacct
+                    job_state = _get_slurm_state_with_sacct(job_id, cluster)
+                    if job_state:
+                        return job_state
+                    return 'NOT_FOUND'
                 raise Exception( '`%s` returned %s, stderr: %s' % ( ' '.join( cmd ), p.returncode, stderr ) )
-            return dict( [ out_param.split( '=', 1 ) for out_param in stdout.split() ] )
+            job_info_dict = dict( [ out_param.split( '=', 1 ) for out_param in stdout.split() ] )
+            return job_info_dict['JobState']
+
         if drmaa_state == self.drmaa_job_states.FAILED:
             try:
-                job_info = __get_jobinfo()
+                slurm_state = _get_slurm_state()
                 sleep = 1
-                while job_info['JobState'] == 'COMPLETING':
+                while slurm_state == 'COMPLETING':
                     log.debug( '(%s/%s) Waiting %s seconds for failed job to exit COMPLETING state for post-mortem', ajs.job_wrapper.get_id_tag(), ajs.job_id, sleep )
                     time.sleep( sleep )
                     sleep *= 2
                     if sleep > 64:
                         ajs.fail_message = "This job failed and the system timed out while trying to determine the cause of the failure."
                         break
-                    job_info = __get_jobinfo()
-                if job_info['JobState'] == 'NOT_FOUND':
+                    slurm_state = _get_slurm_state()
+                if slurm_state == 'NOT_FOUND':
                     log.warning( '(%s/%s) Job not found, assuming job check exceeded MinJobAge and completing as successful', ajs.job_wrapper.get_id_tag(), ajs.job_id )
```

Comment:
With the new `sacct` check this message isn't entirely accurate, but the logic is probably still correct (since not all sites may have accounting configured properly)